### PR TITLE
apt-upgrades: add pending upgrades script

### PR DESCRIPTION
**Pull request by kb100**

### CODE REVIEW

**Comment by kb100** [comment]
I don't know why it shows all those commits that are already in the i3blocks-contrib timeline... did I mess up my git commands?

**Comment by Zopieux** [comment]
Would you mind using a more specific name for this script, as it meant to work with *apt*? Something like `apt-upgrades`. Others may then contribute `pacman-upgrades`, `yum-upgrades` and the likes.

**Comment by Zopieux** [comment]
@kb100 You way want to `git rebase -i <common parent commit>` and force-push the result to remove (stash) all the unnecessary commits. See [this little guide](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#how-do-i-rebase).

**Comment by kb100** [comment]
Alright, I've sorted out my git. Please have a close look now.

File: apt-upgrades/apt-upgrades [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        c) REFRESH_COLOR="$OPTARG" ;;
+        n) NONREFRESH_COLOR="$OPTARG" ;;
+        h) printf "Usage: apt-upgrades [-s refresh_symbol] [-o] [-c refresh_color] [-n nonrefresh_color] [-h]\nOptions:\n-s\trefresh symbol\n-o\tshow symbol only, no numbers\n-c\tcolor when upgrade pending\n-n\tcolor when no upgrade pending\n-h\tshow this help text\n" && exit 0;;
+    esac
+done
+
+read upgraded new removed held < <(
+aptitude full-upgrade --simulate --assume-yes |\
+    grep -m1 '^[0-9]\+ packages upgraded,' |\
+    tr -cd '[0-9 ]' |\
```
Comment (Zopieux) [inline apt-upgrades/apt-upgrades]: > [SC2021](https://github.com/koalaman/shellcheck/blob/0f00de80fd1dd16c21b603625d90fa92795ca22f/ShellCheck/Analytics.hs#L1640) Don't use [] around ranges in tr, it replaces literal square brackets.

I suggest passing your shell scripts in [Shellcheck](http://www.shellcheck.net/) which is an amazing tool to statically analyse common pitfalls. Indeed, if you put `[]` in `tr`, they're not removed:
```shell
$ echo 'foo 73 [bar 112]' | tr -cd '[0-9 ]'
 73 [ 112]
```

**Comment by Zopieux** [comment]
Except for the little inline comment I made, I believe the PR is fine. Thanks for contributing!

**Comment by kb100** [comment]
@Zopieux nice find, fix merged and rebased.

**Comment by vivien** [comment]
That's one clean commit! Thanks to both of you :-)

**Pull request merged**